server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -252,7 +234,7 @@ private void allocateUnassigned() {
                                     }
                                     continue nextShard;
                                 }
-                                case THROTTLE -> isThrottled[0] = true;
+                                case THROTTLE -> isThrottled.set(true);
```

Comment:
Hmm I don't think this needs the visibility guarantees of an `AtomicBoolean`, we only use it on this one thread.